homeassistant/components/habitica/button.py

```diff
@@ -86,6 +225,25 @@ async def async_setup_entry(
     async_add_entities(
         HabiticaButton(coordinator, description) for description in BUTTON_DESCRIPTIONS
     )
+    if coordinator.data.user["stats"]["lvl"] >= 10:
+        if coordinator.data.user["stats"]["class"] == MAGE:
+            async_add_entities(
+                HabiticaButton(coordinator, description) for description in MAGE_SKILLS
+            )
+        if coordinator.data.user["stats"]["class"] == WARRIOR:
+            async_add_entities(
+                HabiticaButton(coordinator, description)
+                for description in WARRIOR_SKILLS
+            )
+        if coordinator.data.user["stats"]["class"] == ROGUE:
+            async_add_entities(
+                HabiticaButton(coordinator, description) for description in ROGUE_SKILLS
+            )
+        if coordinator.data.user["stats"]["class"] == HEALER:
+            async_add_entities(
+                HabiticaButton(coordinator, description)
+                for description in HEALER_SKILLS
+            )
```

Comment:
we should collect all entities and add them once. 

Also, would it be logical to create logic to compare the state of the user on runtime? We could register a listener at the coordinator and check the class of the user and compare that with the class we added entities for, and if they differ delete the entities for that other class. This would also enable us to auto add on reaching level 10.

Not a blocker, but maybe a nice way to give the intergation that extra ✨

Reply:
Would love to do that but don't know how, you surely know an integration to get some inspiration from 😬

Reply:
`airgradient`